src/libtiled/object.cpp

```diff
@@ -40,6 +40,19 @@ ObjectTypes Object::mObjectTypes;
 Object::~Object()
 {}
 
+void Object::mergeComponents(const Components &components)
+{
+    QMapIterator<QString, Properties> it(components);
+    while (it.hasNext()) {
+        it.next();
+        auto const &name = it.key();
+        auto const &value = it.value();
+
+        // overwrites everything
```

Comment:
I don't think this comment is necessary. If you want to explain the behavior of `mergeComponents` you can add a doc comment there. But I think this short explanation is misleading, since it only overwrites the values of the properties defined in `value`, while keeping any other property values.